sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -15,66 +15,46 @@
 # with this program. If not, see <http://www.gnu.org/licenses/>.
 
 import rospy
-import time
 import speech_recognition as sr
-from difflib import get_close_matches
 from std_msgs.msg import String
 
 
 class SpeechControl(object):
-    def __init__(self, trigger_word, command_words, command_topic='speech_control',
-                 non_speaking_duration=0.1, pause_threshold=0.2):
+    def __init__(self, trigger_word, command_topic='sr_speech_control'):
+        self.microphone = sr.Microphone()
+        for i, microphone_name in enumerate(sr.Microphone.list_microphone_names()):
+            if "pulse" in microphone_name:
+                self.microphone = sr.Microphone(device_index=i)
+                rospy.loginfo("Using microphone: {}".format(microphone_name))
         self.trigger_word = trigger_word
-        self.command_words = command_words
         self.recognizer = sr.Recognizer()
-        self.microphone = sr.Microphone()
         self.command_publisher = rospy.Publisher(command_topic, String, queue_size=1)
-        self.command_to_be_executed = None
-
-        self._init_recognizer(non_speaking_duration, pause_threshold)
         self._stop_listening = self.recognizer.listen_in_background(self.microphone, self._recognizer_callback)
 
-    def _init_recognizer(self, non_speaking_duration, pause_threshold):
-        with self.microphone as source:
-            self.recognizer.adjust_for_ambient_noise(source)
-        self.recognizer.non_speaking_duration = non_speaking_duration
-        self.recognizer.pause_threshold = pause_threshold
-
     def _recognizer_callback(self, recognizer, audio):
         try:
             result = recognizer.recognize_google(audio)
+            rospy.loginfo("Recognition result: {}".format(result))
```

Comment:
Personally, I disagree with this. It will only polute the output with unnecessary info about recognized speech that has nothing to do with the commands, but the user only cares if a desired command is recognized and executed. It's ok for debugging, but as far as I'm concerned, not for actual use.

Reply:
It was meant for debugging only as I had a lot of problems. Removed in 3c1f98b6ec65baeb85e9180f487abea57c72b6e0.
It would be bad idea to log this for privacy concerns anyway as it would record everything clients would talk during operation.

Reply:
:+1: